packages/dds/tree/src/test/tree/anchorSet.spec.ts

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { Anchor, AnchorSet, clonePath, Delta, FieldKey, JsonableTree, UpPath } from "../../tree";
+import { brand } from "../../util";
+
+const fieldFoo: FieldKey = brand("foo");
+const fieldBar: FieldKey = brand("bar");
+const fieldBaz: FieldKey = brand("baz");
+const node: JsonableTree = { type: brand("A"), value: "X" };
+
+const path1 = makePath([fieldFoo, 5], [fieldBar, 4]);
+const path2 = makePath([fieldFoo, 3], [fieldBaz, 2]);
+const path3 = makePath([fieldFoo, 4]);
+
+describe("AnchorSet", () => {
+    it("preserves paths", () => {
+        const [anchors, anchor1, anchor2, anchor3] = setup();
+        checkEquality(anchors.locate(anchor1), path1);
+        checkEquality(anchors.locate(anchor2), path2);
+        checkEquality(anchors.locate(anchor3), path3);
+    });
+
+    it("can rebase over insert", () => {
+        const [anchors, anchor1, anchor2, anchor3] = setup();
+
+        const insert = {
+            type: Delta.MarkType.Insert,
+            content: [node, node],
+        };
+
+        anchors.applyDelta(makeDelta(insert, makePath([fieldFoo, 3])));
```

Comment:
It would be good to test at an index such that some anchors are before it and some are after